Gems/Atom/RHI/Vulkan/Code/Source/RHI/SwapChain.cpp

```diff
@@ -442,10 +477,18 @@ namespace AZ
                 imageAvailableSemaphore->GetNativeSemaphore(),
                 VK_NULL_HANDLE,
                 acquiredImageIndex);
+
+            RHI::ResultCode result = ConvertResult(vkResult);
+
+            // The image acquisition can fail if the requested vsync interval exceeds 1.
+            if (vkResult == VK_NOT_READY)
+            {
+                RETURN_RESULT_IF_UNSUCCESSFUL(result);
+            }
```

Comment:
This should be an error. If not the wait and signal semaphores will not be properly handled.

Reply:
Hmm I didn't think I needed to handle this because of the code below. How is this handled in production further down if the assert isn't triggered?

Reply:
I'll need to study the spec more carefully. `VK_NOT_READY` is actually listed as a "success" case [here](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkAcquireNextImageKHR.html) but nothing in the spec seems to suggest what exactly this means. Thanks for the comments though I think I understand the synchronization model used here a bit more.